### PR TITLE
feat: add OTP flow for email verification

### PR DESCRIPTION
**Pull request by owlas**
Closes: #4635 

Blocked by: #4648 

### Description:

- Adds OTP table to db
- Adds models manage OTPs
- New endpoint to trigger OTP emails

### To test

See `examples/api/verify-emails.http` to:

- login
- check your verification status
- trigger a new top
- check verification status again
- try and verify with a passcode

### CODE REVIEW

File: packages/backend/src/database/migrations/20230301161351_add_email_otp_table.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+    if (!(await knex.schema.hasTable('email_one_time_passcodes'))) {
+        await knex.schema.createTable('email_one_time_passcodes', (table) => {
+            table
+                .increments('email_id')
```
Comment (ZeRego) [inline packages/backend/src/database/migrations/20230301161351_add_email_otp_table.ts:7]: What is the purpose of the increment?

File: packages/backend/src/database/entities/email_one_time_passcodes.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    number_of_attempts: number;
+};
+
+export type DbEmailOneTimePasscodeIn = Pick<
+    DbEmailOneTimePasscode,
+    'email_id' | 'passcode'
+>;
+export type EmailOneTimePasscodeTable = Knex.CompositeTableType<
+    DbEmailOneTimePasscode,
+    DbEmailOneTimePasscodeIn
```
Comment (ZeRego) [inline packages/backend/src/database/entities/email_one_time_passcodes.ts:18]: Do we want to have a different update type?

File: packages/backend/src/database/migrations/20230301161351_add_email_otp_table.ts (hunk 1 above)
Comment (owlas) [reply packages/backend/src/database/migrations/20230301161351_add_email_otp_table.ts:7]: this is a mistake!

File: packages/backend/src/models/EmailModel.ts [hunk 3]
```diff
@@ -1,7 +1,27 @@
...
+type DbEmailStatus = Pick<DbEmail, 'email' | 'is_verified'> &
+    Partial<DbEmailOneTimePasscode>;
+
+const convertEmailStatusRow = (row: DbEmailStatus): EmailStatus => {
+    const emailStatus: EmailStatus = {
+        email: row.email,
+        isVerified: row.is_verified,
+        otp: undefined,
+    };
+    console.log(row);
```
Comment (ZeRego) [inline packages/backend/src/models/EmailModel.ts:16]: todo: remove

**Review by ZeRego — APPROVED** [review]
lgtm :) just had to fix the return status on otp creation

**Pull request merged**